src/containers/audio-rewards-page/components/ClaimRewardButton.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+
+import { useSelector } from 'react-redux'
+
+import { useScript } from 'hooks/useScript'
+import AudiusBackend from 'services/AudiusBackend'
+import { getUserHandle } from 'store/account/selectors'
+import { COGNITO_SCRIPT_URL } from 'utils/constants'
+
+// this button will handle triggering the cognito flow
+// or showing the HCaptchaModal
+// or both
+// logic for which flow exactly to trigger is TBD
+// pending AAO changes and DP undisbursed challenges integration
+const ClaimRewardButton = () => {
+  const handle = useSelector(getUserHandle)
+  const scriptLoaded = useScript(COGNITO_SCRIPT_URL)
+
+  const handleClick = async () => {
+    const { signature } = await AudiusBackend.getCognitoSignature()
+
+    // @ts-ignore
+    const flow = new Flow({
+      publishableKey: process.env.REACT_APP_COGNITO_KEY,
+      templateId: process.env.REACT_APP_COGNITO_TEMPLATE_ID,
+      user: {
+        customerReference: handle,
+        signature
+      }
+    })
+
+    flow.on('ui', (event: any) => {
+      switch (event.action) {
+        // we can add subsequent logic here
+        // e.g. what happens after the cognito window is "opened" or "closed"
+        default:
+        // nothing
+      }
+    })
+
+    flow.open()
+  }
+
+  return handle && scriptLoaded ? (
+    <button onClick={handleClick}>Claim your reward</button>
```

Comment:
Is there no need to style this? Will styling be on the implementer?

Reply:
Yeah it should be styled. Wasn't sure at the time that design was final but it seems like the button style is not likely to change. Can confirm with Julian and add styling. Should probably also externalize the click handler. This will be more clear as we get the undisbursed logic in the client.

Reply:
It should get tree-shaken out for now since it's not used anywhere. ok to merge as is and figure out how to consolidate this component w/ the UI button